lang/lfortran/Portfile

```diff
@@ -86,4 +87,19 @@ if {${configure.build_arch} ni [list ppc ppc64]} {
 
 cmake.build_type        Release
 
-test.run                yes
+variant kokkos description "Enable Kokkos support" {
+    depends_lib-append  path:lib/libkokkoscore.dylib:kokkos
+
+    configure.args-append \
+                        -DWITH_KOKKOS=YES
+}
+
+default_variants-append +kokkos
+
+# FIXME: test_lfortran fails at the moment.
+# MacPorts CI are broken and force running tests
+# even when those are disabled. Therefore this ugly hack:
+test {
+    ui_info "Tests are disabled."
+    return
+}
```

Comment:
why does this only appear to be a problem for some of your ports? The CI works perfectly fine for 99% of the cases...

Reply:
Well, we know why was that the issue with OCaml and R ports, but I have no idea why it happens here, since the port does not use custom PG overriding test phase.

Reply:
@jmroot -- a ruling please.

@barracuda156 will not update any of his ports to use MacPorts currently designated python 3.12, because python 3.12 doesn't work on 10.6 PPC.

@reneeotten  and others keep asking him to do so, which is getting tedious.

What would you like done?

Reply:
@kencu If you or anyone with commit access want to modify my ports, I have no way to prevent that. But indeed, don’t make me involved, please. I am not gonna break my own ports for no good reason.
Somebody else can break them, but no one can force me to do that. This is not my employment, after all, I have no such an obligation.

If a change _fixes nothing_ but _breaks something,_ it should not be done. Especially without testing. Simple as that. Even if it breaks a system with three users. It still improves nothing.
Does `lpython` upstream say that Python 3.11 cannot work with it? That will be a credible reason to switch it, of course. But not anyone’s personal preferences, sorry.

Instead of wasting time to force switch a working Python for a less working one, maybe you could fix a broken GCC in Macports? What is now Macports’s default (13.2.0) was made such without proper testing, and it does not work with `libc++` (yeah, 10.6 ppc and Leopard are fine, while Sonoma is broken).
This is by the way an illustration why blind updates of this sort can be harmful. But no one cares, this is broken for months. Much more important is to change default Python, and specifically in my ports, and not hundreds of others which also use older versions.

Reply:
Of course, I don't want to modify anyone's ports, and I don't want to have to talk about this issue at all ever again! 

So let's just get a ruling, and have this over with, whatever it is, so people stop having friction about this one thing. Josh may well say you can do what you want with your 3000+ ports and keep them all at python 3.11, etc. But we don't want to keep bring this up every single PR like it has been.

Reply:
@kencu By the way, if you had in mind to make a genuine improvement, here is one way to do it, satisfying all sides, perhaps:

Have an umbrella port for Python to run scripts (it is just build system and/or running tests, any Python would do), and then you can set whatever you prefer for Intel and arm64, and I can have my powerpc-only port, which is used only on PowerPC systems. Everyone is happy then?

Reply:
Josh set a policy that all ports use MacPorts designated python (currently 3.12) unless the port cannot use it.

I didn't set the policy (although I agree with it).

Don't take it so personally -- if it is the policy, it's the policy. Abide by it.

Reply:
The policy is phrased as a "should" not a "must", but yes, please don't hold newer systems hostage to the needs of old ones. In this case it should not be hard to use 3.12 when it works and a different version when it doesn't.

Reply:
I too stopped blindly updating ports to use python 3.12 because we already had a couple instances where doing so broke ports (e.g. https://trac.macports.org/ticket/69036 and https://trac.macports.org/ticket/69228) and I am not a python expert and am not interested in investigating how to fix those breakages. I leave it to someone else to do in many cases.

> CI should be fixed to respect `test.run no`

MacPorts base was [enhanced in 2.9.0](https://github.com/macports/macports-base/commit/ddc26554e6ed49ce8501069b967284739fa6d34e) to have a built-in test method that is used for all ports, even those that do not set `test.run yes`. CI was [deliberately changed](https://github.com/macports/macports-ports/commit/d3219b762afe61e965be308d1a165f35c944abbd) to run those built-in tests. There is a bug in MacPorts base that causes the built-in tests to fail if a file is unreadable. This will be [fixed in MacPorts 2.9.2](https://github.com/macports/macports-base/commit/e3468c5d60b9a5872d37b57c3d0e6a60694daac9).

Reply:
OK, so I would take that to mean that the PR reviewers who keep pointing out to @barracuda156 that he is meant to use python 3.12 can stop doing so.

That should improve everyone’s interactions nicely, and we can move on back to the fun stuff.

Reply:
to me it read as "yes, use Python 3.12 wherever it works and if you really want to restrict old systems to an earlier version". But I will just stop asking and/or interacting with PRs like this and leave it up to whoever wants to merge things. 

I agree of course that blindly updating to Python 3.12 should not be done, but I'm pretty sure the majority of the ports will work just fine with Python 3.12.

Reply:
@jmroot Thank you, Josh, noted.

@reneeotten As you well know, I do not avoid python 3.12 everywhere out of principle, and normally pro-actively add it to Python ports. But there is really no urgency to rush a switch for non-Python ports which use Python for technical needs, as long as existing choice works for all supported systems.

Reply:
That is not really true, though.

One of the largest complaints agains MacPorts as compared to homebrew is bloat, and many people find they need to have multiple versions of perl, python, llvm, clang, and other things installed.

People leave MacPorts for this exact reason.

So to address those complaints, MacPorts agreed to standardize on one python version (and perl version) to be used above all others, and only use another version if there was some good reason to do so (like the recommended version doesn't work).

So by your forcing everyone to keep python3.11 around to suit your 10.6-PPC purposes, you do indeed cause needless bloat and worsen (and undermine) that process.

Reply:
However, the admins don't want to be too dictatorial or chase you away, so they acquiesce to this, solely and uniquely for you, pretty much.

Reply:
@kencu 
> One of the largest complaints agains MacPorts as compared to homebrew is bloat, and many people find they need to have multiple versions of perl, python, llvm, clang, and other things installed.

_This is very well beyond the scope of present discussion._
I see advantages and disadvantages with either of approaches, but I am pretty sure that the market does not need another identical product.
Perhaps with modern hardware and internet speed (and pre-built ports) the problem is more imaginary than practical. For me on a PowerMac it is a pain that Macports needs to build 4–5 empty libgcc ports which takes a full day with strictly none positive utility, but with pre-built ports on arm64 we have much more versatility, power and compatibility at the cost of few extra minutes of downloading and occasionally compiling something.

> So by your forcing everyone to keep python3.11 around to suit your 10.6-PPC purposes, you do indeed cause needless bloat and worsen (and undermine) that process.

At least at the moment there are numerous ports defaulting to 3.11 and even earlier, so I don’t think right now this increases havoc.

I get everyone’s points, let’s really save our time for a more productive topic, which we actually have. (And I hope we find a neat solution for Pythons too.)
We will not keep it stuck at 3.11 indefinitely in any case.

Reply:
I don't think you do get everyone's points, and this is a very productive topic -- needlessly forcing everyone to stay on python 3.11 solely so you can make a point about getting your way supporting 10.6 for PPC is very selfish.